lapce-ui/src/completion.rs

```diff
@@ -60,19 +61,48 @@ impl Snippet {
         (elements, pos)
     }
 
-    fn extract_tabstop(s: &str, pos: usize) -> Option<(SnippetElement, usize)> {
-        for re in &[
-            Regex::new(r#"^\$(\d+)"#).unwrap(),
-            Regex::new(r#"^\$\{(\d+)\}"#).unwrap(),
-        ] {
-            if let Some(caps) = re.captures(&s[pos..]) {
-                let end = pos + re.find(&s[pos..])?.end();
-                let m = caps.get(1)?;
-                let n = m.as_str().parse::<usize>().ok()?;
-                return Some((SnippetElement::Tabstop(n), end));
-            }
+    fn extract_tabstop(str: &str, pos: usize) -> Option<(SnippetElement, usize)> {
```

Comment:
Maybe remove this and just use the one in lapce-data/src/completion.rs

Reply:
Since this is an associated function of the `struct Snippet`, this requires removing the `struct Snippet` completely from
`lapce-ui/src/completion.rs`, and use the `struct Snippet` definition from `lapce-data/src/completion.rs` in `lapce-ui`.

However, I think this is not difficult, although I probably have to remove the definition for `SnippetElement` as well. Do you want it?

Reply:
Yeah the structs should be removed as they look duplicate as well.

Reply:
The definition of `struct Snippet` has been completely removed from `lapce-ui/src/completion.rs`, as this structure is surprisingly **not used** at all in **`lapce-ui`**.